src/core/lib/security/authorization/rbac_translator.cc

```diff
@@ -104,6 +104,12 @@ absl::StatusOr<Rbac::Principal> ParsePeer(const Json& json) {
     if (it->second.type() != Json::Type::ARRAY) {
       return absl::InvalidArgumentError("\"principals\" is not an array.");
     }
+    if (it->second.array_value().empty()) {
+      return Rbac::Principal(
+          Rbac::Principal::RuleType::kPrincipalName,
+          StringMatcher::Create(StringMatcher::Type::kExact, /*matcher=*/"")
```

Comment:
Returning a string matcher that exactly matches the empty string seems very strange here.  Why does this work?  Wouldn't this wind up denying access if the authenticated principal is a non-empty string?  Is there some more intuitive way of representing this rule?

Reply:
Based on the discussion in proposal
https://github.com/grpc/proposal/pull/246#discussion_r758805134
I made some updates to the PR. Removed the special-case for empty principals `principals: []`.
To allow authenticated connection, we can do `principals: ["*"]`